feat(color): add color contrast helpers

souporserious opened the pull request:
Adds color contrast helpers using calculations from W3C.

closes #394

bhough commented on src/types/color.js:
```diff
@@ -3,6 +3,19 @@
+ * @property {number}  AAA
+ * @property {number}  AAALarge
+ */
+export type ContrastScores = {|
```
Since this is being used as a return type for `getColorContrastScores` each key should return `boolean` as opposed to `number`. This should fix the failing Flow tests in the build.

souporserious replied:
Ah, totally! I'll get this fixed up.

bhough commented on src/color/getColorContrast.js:
```diff
@@ -0,0 +1,23 @@
+ * @example
+ * const contrastRatio = getColorContrast('#444', '#fff');
+ */
+export default function getColorContrast(
```
Wondering if we should just call this `getContrast` to stay in line with `getLuminance`.

bhough commented on src/color/getColorContrastScores.js:
```diff
@@ -0,0 +1,24 @@
+ * @example
+ * const scores = getColorContrastScores('#444', '#fff');
+ */
+export default function getColorContrastScores(
```
Wondering if this something like `meetsContrastGuidelines` might be better since we really aren't returning scores as much as checking if they comply with the W3s guidelines for different scenarios.

souporserious replied:
Yeah, I like that name way better! 😄

bhough commented on src/color/test/getColorContrast.test.js:
```diff
@@ -0,0 +1,42 @@
+    ).toMatchSnapshot()
+  })
+
+  it('should return the color contrast of two named CSS colors', () => {
```
One case not tested (and worth discussing) is how these two modules should return if passed `transparent` as one (or both) of the colors. `getLuminance` returns 0 in these cases, so may be as simple as just testing if we are ok with that. Obviously an edge case, but we do account for it in most color modules.

souporserious replied:
Yeah, I wasn't sure how to approach transparency since I believe it could give a false-positive when comparing something like `black` and `transparent`. I'll add some tests around it and then I'd be happy to discuss it further.

bhough commented on src/color/test/getColorContrastScores.test.js:
```diff
@@ -0,0 +1,47 @@
+    ).toMatchSnapshot()
+  })
+
+  it('should return the color contrast scores of two named CSS colors', () => {
```
See above about `transparent` use case.

bhough commented on src/color/getColorContrastScores.js:
```diff
@@ -0,0 +1,24 @@
+import type { ContrastScores } from '../types/color'
+
+/**
+ * Returns the contrast scores between two colors based on the
```
Should update the description since it is not returning scores but instead checking the codes compliance with various guideline values.

souporserious commented:
Thanks for the feedback @bhough! I'll get these fixed up this week.

souporserious commented:
@bhough addressed all of your comments. Hopefully, everything should be good now 🤞. Please let me know if there is anything else you see that needs changing.

bhough commented:
Thanks @souporserious! I will give this all a look this weekend and hopefully get a release out.
